### PR TITLE
Support send push notification by topic

### PR DESCRIPTION
**Pull request by ben181231**
connect SkygearIO/skygear-server#239

### CODE REVIEW

File: Pod/Classes/Operations/SKYRegisterDeviceOperation.h [hunk 1]
```diff
@@ -22,7 +22,9 @@
 
 @interface SKYRegisterDeviceOperation : SKYOperation
 
-- (instancetype)initWithDeviceToken:(NSData *)deviceToken;
+- (instancetype)initWithDeviceToken:(NSData *)deviceToken
+    __attribute__((deprecated("Use -(void)initWithDeviceToken:topic: instead")));
```
Comment (cheungpat) [inline Pod/Classes/Operations/SKYRegisterDeviceOperation.h]: interesting, i didn’t know about this
no need for the return type of the method here, and btw the return type of the method is incorrect.

Ditto for below.

File: Example/Swift Example/AppDelegate.swift [hunk 2]
```diff
@@ -36,28 +66,32 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+    func application(application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: NSData) {
+        print("Successfully registered remote notification")
+        self.registerDevice(deviceToken)
     }
 
-    func applicationWillTerminate(application: UIApplication) {
-        // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+    func application(application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: NSError) {
+        print("Failed to register remote notification: \(error.localizedDescription)")
+        self.registerDevice()
```
Comment (cheungpat) [inline Example/Swift Example/AppDelegate.swift]: I would suggest to inline the `registerDevice` functions in the app delegate functions. That way the developer knows what to do in app delegate without referencing other functions.

Comment (ben181231) [inline Example/Swift Example/AppDelegate.swift]: Okay.

`registerDevice()` is registering a device to skygear with `null` device token. The device ID will be reused once the SDK can get the device token.

Comment (cheungpat) [inline Example/Swift Example/AppDelegate.swift]: Still not sure what you mean, waiting for your update to this PR

**Comment by ben181231** [comment]
Updated.

**Comment by ben181231** [comment]
Updated.

**Pull request merged**